api/envoy/extensions/filters/http/ip_tagging/v3/ip_tagging.proto

```diff
@@ -39,6 +40,20 @@ message IPTagging {
     EXTERNAL = 2;
   }
 
+  // Describes how to apply the tags to the headers.
+  enum HeaderAction {
+    // (DEFAULT) The header specified in :ref:`ip_tag_header <envoy_v3_api_field_extensions.filters.http.ip_tagging.v3.IPTagging.ip_tag_header>`
+    // will be dropped, before the tags are applied.
+    SANITIZE = 0;
```

Comment:
So AFIK most x-envoy headers (so default ip tagging headers) are sanitized iff the downstream endpoint is untrusted and not sanitized if sent from a trusted IP.

It looks from a quick glance at the code like the new filter option unilaterally sanitizes.  if this is the goal, I think we should call out the difference here, lest someone assume sanitize is normal "do we trust downstream" sanitization

Reply:
Also regardless of if we force-sanitize or x-envoy sanitize, the header will be visible unsanitized to any filters downstream of the ip-tag-header filter.  That should be obvious but again I think it's worth being verbose and calling out differences.

Reply:
@alyssawilk 
> Also regardless of if we force-sanitize or x-envoy sanitize, the header will be visible unsanitized to any filters downstream of the ip-tag-header filter.

I don't think I understand what you mean here.
Far as I understand, once the ip-tagging-filter has run, any filters after it within the chain will only see the changes made by ip-tagging-filter, and not see any preceding state.

Reply:
oh, I'm swapping around the meaning of upstream and downstream. It makes sense. <3